tests/TestCase/TestSuite/TestCaseTest.php

```diff
@@ -146,6 +146,32 @@ public function testWithErrorReportingWithException(): void
         }
     }
 
+    /**
+     * test deprecated
+     */
+    public function testDeprecated(): void
+    {
+        $this->deprecated(function () {
+            deprecationWarning('test');
+        });
+    }
+
+    /**
+     * test deprecated
+     */
+    public function testDeprecatedWithNoDeprecation(): void
+    {
+        $message = '';
+        try {
+            $this->deprecated(function () {
+            });
```

Comment:
conflict when passing ```deprecationWarning('test');``` same as previous test `testDeprecated`

but ```deprecationWarning('other string');``` will stop assert failing in ` $this->assertTrue($deprecation, ...`

Reply:
this was new feature
Refs: https://github.com/cakephp/cakephp/commit/eef535c1153a9